### ISSUE TITLE
some python syntax error in tinydb's document

### ISSUE THREAD

**Issue by licovery** [msiemens/tinydb#199]
https://tinydb.readthedocs.io/en/latest/usage.html#replacing-data

```
>>> docs = db.search(User.name == 'John')
[{name: 'John', age: 12}, {name: 'John', age: 44}]
>>> for doc in docs:
...     doc.name = 'Jane'
>>> db.write_back(docs)  # Will update the documents we retrieved
>>> docs = db.search(User.name == 'John')
[]
>>> docs = db.search(User.name == 'Jane')
[{name: 'Jane', age: 12}, {name: 'Jane', age: 44}]
```

**I am puzzled by this:**
```
>>> for doc in docs:
...     doc.name = 'Jane'
```

It seems that the content of docs does not change at all.
I am not sure I'm right, I'm not an expert in python.

**Comment by licovery** [comment]
it words when I use [ ] 

```
>>> for doc in docs:
...     doc['name'] = 'Jane'
```

**Comment by licovery** [comment]
```
>>> for doc in docs:
...     print(doc.name)
```
Python come up with an error
AttributeError: 'Document' object has no attribute 'name'

**Comment by Phuket2** [comment]
Its a bit of a digression, but below I did a test inheriting from Tinydb's Document class.  Just interesting for me at least. It does go to idea of being able to specify your own Document class to tinydb though.  If you inherit from the Document class, call the super and was able too assign your subclassed class to tinydb as the Document, would be no need to do what I did below after reading the data back in from the db (create new objects).  I am not sure if you could do the same with custom storage/middleware.  Anyway, Its just an idea. I haven't tried to use it for anything practical yet, just simple tests.  

```python
from tinydb import TinyDB
from tinydb.database import Document

from datetime import datetime as datetime

from faker import Faker
fake = Faker()

class PersonClass(Document):
    '''
    value is a dict
    
    doc_id is used internally by tinydb.  But we need to pass an Integer
    to be able to call the super.__init__ on the Document class.
    When creating these objects(from sratch), I am passing any int, could be 0.
    When creating these objects from data read back from the db, the doc_id,
    has some meaning as long as you pass the doc_id from the recovered database
    doc. 
    '''
    def __init__(self, value, doc_id, **kwargs):
        super().__init__(value, doc_id, **kwargs)
    
    @property
    def full_name(self):
        '''
        returns the full name of the person
        '''
        return "{} {}".format(self['first'], self['last'])
    
    @property
    def some_date_dt(self):
        '''
        The date stored in the db is a string as you have to.
        Just getting a datetime object here. Could help with
        serialisation if you dont need generic code. 
        '''
        the_date = datetime.strptime(self['some_date'], "%Y-%m-%d")
        return the_date
        
    @classmethod
    def from_tinydb_doc(cls, obj):
        '''
        Create a PersonClass from a tinydb doc obj
        Not required for this simple test, but I am just practicing
        doing it.
        '''
        return PersonClass(obj, obj.doc_id)         
    

def make_dummy_people(num_items=100):
    '''
    Create some dummy people to use...
    returns a list of PersonClass objects 
    The data built up using Faker
    '''
    
    return [
            PersonClass(
                {'idx': i,
                'first': fake.first_name(),
                'last': fake.last_name(),
                'age': fake.random_int(min=18, max=106),
                'some_date':fake.date(pattern="%Y-%m-%d"),
                },
                i)
            for i in range(0, num_items)
            ]


db_fn = 'doc_test.json'                  # database filename
dummy_data = make_dummy_people(10)

db = TinyDB(db_fn)                       # create TDB instance w/filename
db.purge_tables()                        # clear all the tables, just for testing
db.insert_multiple(dummy_data)           # add the dummy data to the new table

for doc in db.all():
    # creating PersonClass objects from the doc obj's
    # returned from tinydb.  Its double work, but could be worth it
    # for flexibilty.  With this simple example, very easy to re-create
    # your own document class from the db.
    
    # below line is commented, can use a normal creation or the
    # from_tinydb_doc class method below.  Just trying different things
    #person_doc = PersonClass(doc, doc.doc_id)
    
    # try a classmethod, to create the PersonClass
    person_doc = PersonClass.from_tinydb_doc(doc)
    
    # just print some stuff to show it works
    print(person_doc.doc_id, person_doc.full_name, person_doc.some_date_dt)
    
db.close()

```

**Comment by msiemens** [comment]
Seems like I got a little to much used to writing JavaScript/TypeScript lately where dot access works with objects 🙈 Anyway, the docs have been fixed in #205 🙂
